File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.cc
#include <iostream>
#include <cmath>

using namespace std;

int gcd(int a, int b){
    int c = a % b;
    while(c != 0){
        a = b;
        b = c;
        c = a % b;
    }
    return b;
}

int lcm(int a, int b){
    return (a*b)/ gcd(a,b);
}
int main()
{   
    int M, N;
    cin >> N >> M;
    
    cout << gcd(N, M) << endl;
    cout <<lcm(N, M);
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.util.*;

class Solution {
    public int[] solution(int n, int m) {
        int[] answer = new int[2];

        int max = Math.max(n, m);
        int min = Math.min(n, m);
        
        answer[0] = gcd(max, min);
        answer[1] = lcm(max, min);
        
        return answer;
    }
    
    public int gcd(int a, int b){
        if(a%b==0){
            return b;
        }
        
        return gcd(b, a%b);
    }
    
    public int lcm(int a, int b){
        int g = gcd(a,b);
        
        return a*b/g;
    }
}